src/app/currency.pipe.ts

```diff
@@ -0,0 +1,25 @@
+import { Pipe, PipeTransform, Injectable } from '@angular/core';
+import { Currency } from './models/budget-calculator.model';
+
+@Injectable({ providedIn: 'root' })
+@Pipe({
+  name: 'convert',
+  standalone: true,
+})
+export class CurrencyConvertPipe implements PipeTransform {
+  transform(value: number, currency: Currency, convertToCurrency: Currency) {
+    if (!value || isNaN(value)) {
+      return null;
+    }
+    // console.log(value, currency, convertToCurrency);
+    if (currency === convertToCurrency) {
+      return value.toFixed(2);
+    } else if (currency === 'USD' && convertToCurrency === 'BGN') {
+      return (value * 1.75).toFixed(2);
```

Comment:
Numbers such as `1.75` and `0.57` can be extracted into a separate file called:
- `conversion-rates.ts`

```ts
export const USD_TO_BGN_CONVERSATION_RATE = 1.75
export const BGN_TO_USD_CONVERSION_RATE = 0.57
```
This will increase code readability.

@TheaAleksovDev - would you be interested If I were to create a new issue to enhance the Application to fetch the rates from an REST API? This will be a new task, where you will implement the [REST polling pattern](https://medium.com/weekly-webtips/polling-in-angular-how-to-use-rxjs-in-angular-for-polling-14c519f4c218).
Basically fetching currency rates every minute or SO.

Reply:
That sounds interesting. Yes, I'd like another challenge.

Reply:
![nice](https://github.com/user-attachments/assets/7d5c47b3-5c27-4c4d-809b-15a6062f5a65)

Here you go:
- https://github.com/TheaAleksovDev/budgetCalculator/issues/6